### ISSUE TITLE
[Bug]: Querynode keeps in terminating status when rolling upgrade from v2.4.3 to master-20240908-e480b103-amd64

### ISSUE THREAD

**Issue by zhuwenxing** [milvus-io/milvus#36100]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: v2.4.3 to master-20240908-e480b103-amd64
- Deployment mode(standalone or cluster):cluster
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

```
2024-09-08T17:41:08.303Z] + + kubectl get pods -o wide

[2024-09-08T17:41:08.304Z] grep kafka-mixcoord-4785

[2024-09-08T17:41:08.304Z] kafka-mixcoord-4785-etcd-0                                       1/1     Running       0               39m     10.104.30.61    4am-node38   <none>           <none>

[2024-09-08T17:41:08.304Z] kafka-mixcoord-4785-etcd-1                                       1/1     Running       0               39m     10.104.34.41    4am-node37   <none>           <none>

[2024-09-08T17:41:08.304Z] kafka-mixcoord-4785-etcd-2                                       1/1     Running       0               39m     10.104.25.44    4am-node30   <none>           <none>

[2024-09-08T17:41:08.304Z] kafka-mixcoord-4785-kafka-0                                      2/2     Running       1 (38m ago)     39m     10.104.30.63    4am-node38   <none>           <none>

[2024-09-08T17:41:08.304Z] kafka-mixcoord-4785-kafka-1                                      2/2     Running       1 (38m ago)     39m     10.104.34.47    4am-node37   <none>           <none>

[2024-09-08T17:41:08.304Z] kafka-mixcoord-4785-kafka-2                                      2/2     Running       1 (38m ago)     39m     10.104.25.49    4am-node30   <none>           <none>

[2024-09-08T17:41:08.304Z] kafka-mixcoord-4785-kafka-exporter-6d94cbb8d8-csk89              1/1     Running       4 (38m ago)     39m     10.104.4.252    4am-node11   <none>           <none>

[2024-09-08T17:41:08.304Z] kafka-mixcoord-4785-kafka-zookeeper-0                            1/1     Running       0               39m     10.104.30.64    4am-node38   <none>           <none>

[2024-09-08T17:41:08.304Z] kafka-mixcoord-4785-kafka-zookeeper-1                            1/1     Running       0               39m     10.104.34.48    4am-node37   <none>           <none>

[2024-09-08T17:41:08.304Z] kafka-mixcoord-4785-kafka-zookeeper-2                            1/1     Running       0               39m     10.104.25.50    4am-node30   <none>           <none>

[2024-09-08T17:41:08.304Z] kafka-mixcoord-4785-milvus-datanode-5f96fdd975-frptg             1/1     Running       0               22m     10.104.6.253    4am-node13   <none>           <none>

[2024-09-08T17:41:08.304Z] kafka-mixcoord-4785-milvus-datanode-5f96fdd975-n92bf             1/1     Running       0               21m     10.104.32.231   4am-node39   <none>           <none>

[2024-09-08T17:41:08.304Z] kafka-mixcoord-4785-milvus-datanode-5f96fdd975-qsdxk             1/1     Running       0               22m     10.104.17.232   4am-node23   <none>           <none>

[2024-09-08T17:41:08.304Z] kafka-mixcoord-4785-milvus-indexnode-7bf75c7445-4bdb8            1/1     Running       0               29m     10.104.34.52    4am-node37   <none>           <none>

[2024-09-08T17:41:08.304Z] kafka-mixcoord-4785-milvus-indexnode-7bf75c7445-jrd72            1/1     Running       0               28m     10.104.6.251    4am-node13   <none>           <none>

[2024-09-08T17:41:08.304Z] kafka-mixcoord-4785-milvus-indexnode-7bf75c7445-s6tc7            1/1     Running       0               30m     10.104.17.223   4am-node23   <none>           <none>

[2024-09-08T17:41:08.304Z] kafka-mixcoord-4785-milvus-mixcoord-8669b565c5-bd2nw             1/1     Running       0               26m     10.104.17.227   4am-node23   <none>           <none>

[2024-09-08T17:41:08.304Z] kafka-mixcoord-4785-milvus-proxy-85987d9957-phdmj                1/1     Running       0               21m     10.104.32.229   4am-node39   <none>           <none>

[2024-09-08T17:41:08.304Z] kafka-mixcoord-4785-milvus-querynode-0-6fcc456958-jbrtn          1/1     Terminating   0               37m     10.104.4.8      4am-node11   <none>           <none>

[2024-09-08T17:41:08.304Z] kafka-mixcoord-4785-milvus-querynode-0-6fcc456958-pgr8z          1/1     Running       0               37m     10.104.1.118    4am-node10   <none>           <none>

[2024-09-08T17:41:08.304Z] kafka-mixcoord-4785-milvus-querynode-1-776f6f9bbf-f9h7k          1/1     Running       0               23m     10.104.34.53    4am-node37   <none>           <none>

[2024-09-08T17:41:08.304Z] kafka-mixcoord-4785-milvus-querynode-1-776f6f9bbf-v2bcf          1/1     Running       0               24m     10.104.17.231   4am-node23   <none>           <none>

[2024-09-08T17:41:08.304Z] kafka-mixcoord-4785-minio-0                                      1/1     Running       0               39m     10.104.30.62    4am-node38   <none>           <none>

[2024-09-08T17:41:08.304Z] kafka-mixcoord-4785-minio-1                                      1/1     Running       0               39m     10.104.34.44    4am-node37   <none>           <none>

[2024-09-08T17:41:08.304Z] kafka-mixcoord-4785-minio-2                                      1/1     Running       0               39m     10.104.25.46    4am-node30   <none>           <none>

[2024-09-08T17:41:08.304Z] kafka-mixcoord-4785-minio-3                                      1/1     Running       0               39m     10.104.21.75    4am-node24   <none>           <none>
```

### Milvus Log

failed job: https://qa-jenkins.milvus.io/blue/organizations/jenkins/rolling_update_for_operator_test_simple/detail/rolling_update_for_operator_test_simple/4785/pipeline
log:
[artifacts-kafka-mixcoord-4785-server-logs.tar.gz](https://github.com/user-attachments/files/16925583/artifacts-kafka-mixcoord-4785-server-logs.tar.gz)

**Comment by weiliu1031** [comment]
please verify this with latest image

**Comment by zhuwenxing** [comment]
verified and fixed in `2.4.14` --> `master-20241105-b83b376c-amd64`
